idseq_dag/steps/run_srst2.py

```diff
@@ -69,6 +72,76 @@ def get_common_params(self):
         return ['--min_coverage', min_cov,'--threads', n_threads,
                 '--output',  os.path.join(self.output_dir_local, 'output'), '--log', '--gene_db', db_file_path]
 
+    def generate_mapped_reads_table(self):
+        """Use bedtools to generate a table of mapped reads for each genome in the ARG ANNOT database.
+            If a new resistance gene db is used, the .bed file will need to be updated manually."""
+        bed_file_path = fetch_from_s3(self.additional_files["resist_genome_bed"], self.output_dir_local, allow_s3mi=False)
+        bedtools_params = ['bedtools', 'coverage', '-b', self.output_files_local()[5], '-a', bed_file_path, '>>', os.path.join(self.output_dir_local, 'matched_reads.tsv')]
+        command.execute(" ".join(bedtools_params))
+
+    def get_total_reads(self, isZipped, isFasta):
+        """Gets the total number of reads in the sample by counting them directly from the
+            fastq or fasta files."""
+        input_filenames = self.input_files_local[0]
+        if isZipped:
+            gunzip_params = ['gunzip', '-k']
+            gunzip_params.extend(input_filenames)
+            command.execute(" ".join(gunzip_params))
+            input_filenames = map(lambda name: name[:len(name)-3], input_filenames)
+        if isFasta:
+            grep_params = ['grep', '-c', '"^>"'] # fastas start reads with "^>"
```

Comment:
Is it not possible to do the `wc -l` approach for fasta files as well?

Reply:
Correct, fasta files vary in the number of lines per read so it would not be possible to calculate the number of reads just from `wc` alone. Fastqs have a set 4 lines per read.

Reply:
Great! I would add a comment here so future people don't wonder the same thing. I didn't realize fasta files can vary in the number of lines per read (I thought it was 2 per read)